### DISCUSSION TITLE
Custom selection area Like default rectangle selection on mouse drag

### DISCUSSION THREAD

**Discussion by ghost** [d3/d3#3591] [Q&A] [answered]
How can we create custom selection area on the graph, like default rectangle selection on mouse drag?
We are looking to allow user to select different shapes of areas on a scatter plot. Shapes like Circle, Triangle and Donut etc.

Kindly guide us with appropriate solution, if applicable.

Reference Image of what is expected to achieve
![image](https://user-images.githubusercontent.com/116060187/196384349-2223687d-1f3a-4d32-a3aa-c37b984954ab.png)
The above picture shows rect on mouse drag, similarly we plan to introduce circle , triangle and other shapes to behave like this.

Please help

**Comment by Fil** [comment, marked as answer]
See [d3-brush](https://github.com/d3/d3-brush) and [d3-lasso](https://observablehq.com/@bumbeishvili/d3-lasso-selection).
